include/Introspection.h

```diff
@@ -0,0 +1,1501 @@
+                if constexpr( std::is_same_v<CurrType,MemberType> ) {
+                    if( Name == Member.GetName() ) {
+                        ToCall(Member);
+                    }
+                }
+            }
+        );
+    }
+
+    /// @brief Checks to see if a class has a named member.
+    /// @tparam Class Provided class type that will be checked.
+    /// @param Name The name of the member to check for.
+    /// @return Returns true of the specified class has a registered member with the provided name, false otherwise.
+    template<typename Class>
+    Boole HasMember(const StringView Name)
+    {
+        Boole Found = false;
+        DoForAllMembers<Class>(
+            [&](const auto& Member) {
+                if( Name == Member.GetName() ) {
+                    Found = true;
+                }
+            }
+        );
+        return Found;
+    }
+
+    /// @brief Convenience method for getting a specific named member on an object.
+    /// @tparam MemberType The type of the member that will be returned.
+    /// @tparam Class The class type to get the member value of.
+    /// @param Obj The object instance to get the value from.
+    /// @param Name The name of the member to retrieve on the object.
+    template<typename MemberType, typename Class>
+    MemberType GetMemberValue(Class& Obj, const StringView Name)
+    {
+        MemberType Value{};
+        DoForMember<Class,MemberType>(Name,
+            [&](const auto& Member) {
+                Value = Member.GetValue(Obj);
+            }
+        );
+        return Value;
+    }
+
+    /// @brief Convenience method for setting the value of a specific named member on an object.
+    /// @tparam MemberType The type of the member that will be returned.
+    /// @tparam Class The class type to set the member value of.
+    /// @tparam ValueType The type of the value to be converted to the named member type.
+    /// @param Obj The object instance to set the value of.
+    /// @param Name The name of the member to set on the object.
+    /// @param Val The value that will be used to set the named member.
+    template<typename MemberType,
+             typename Class,
+             typename ValueType,
+             typename = std::enable_if_t< std::is_convertible_v<ValueType,MemberType> >>
+    void SetMemberValue(Class& Obj, const StringView Name, ValueType&& Val)
+    {
+        DoForMember<Class,MemberType>(Name,
+            [&Obj, &Val = std::forward<ValueType>(Val)] (const auto& Member) {
+                Member.SetValue(Obj,Val);
+            }
+        );
+    }
+
+    /// @page Mezzanine Introspection System
+    /// @section IntrospectOverview Overview
+    /// The Mezzanine Introspection system is a collection of Classes, Structs, and Functions that heavily
+    /// leverage Template Meta-programming to allow generic code for the reading and writing of values on
+    /// classes and/or structs.
```

Comment:
I somewhat disagree about the appearance of Reflection being sooner.  But I have added this to that paragraph:

"With the Introspection system it is possible to create as little as one function to perform the reading and/or writing of values of an arbitrary number of types, rather than creating one or more functions per type."